lib/dependencies/HarmonyImportSpecifierDependency.js

```diff
@@ -2,79 +2,109 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
-
-function HarmonyImportSpecifierDependency(importDependency, importedVar, id, name, range) {
-	NullDependency.call(this);
-	this.importDependency = importDependency;
-	this.importedVar = importedVar;
-	this.id = id;
-	this.name = name;
-	this.range = range;
-}
-module.exports = HarmonyImportSpecifierDependency;
+"use strict";
+const NullDependency = require("./NullDependency");
 
-HarmonyImportSpecifierDependency.prototype = Object.create(NullDependency.prototype);
-HarmonyImportSpecifierDependency.prototype.constructor = HarmonyImportSpecifierDependency;
-HarmonyImportSpecifierDependency.prototype.type = "harmony import specifier";
-
-HarmonyImportSpecifierDependency.prototype.getReference = function() {
-	if(!this.importDependency.module) return null;
-	return {
-		module: this.importDependency.module,
-		importedNames: this.id ? [this.id] : true
-	};
-};
-
-HarmonyImportSpecifierDependency.prototype.getWarnings = function() {
-	var importedModule = this.importDependency.module;
-	if(importedModule && importedModule.meta && importedModule.meta.harmonyModule) {
-		if(this.id && importedModule.isProvided(this.id) === false) {
-			var err = new Error("export '" + this.id + "'" +
-				(this.id !== this.name ? " (imported as '" + this.name + "')" : "") +
-				" was not found in '" + this.importDependency.userRequest + "'");
-			err.hideStack = true;
-			return [
-				err
-			];
+class HarmonyImportSpecifierDependency extends NullDependency {
+	constructor(importDependency, importedVar, id, name, range) {
+		super();
+		this.importDependency = importDependency;
+		this.importedVar = importedVar;
+		this.id = id;
+		this.name = name;
+		this.range = range;
+	}
+
+	get type() {
```

Comment:
This seems to always loose on coverage. When we get to Dependency. Maybe a top level test that tests that any dependencies type can be obtained with `dep.type`.